src/DotNetWorker.ApplicationInsights/FunctionsApplicationInsightsExtensions.cs

```diff
@@ -0,0 +1,85 @@
+        /// <returns>The <see cref="IFunctionsWorkerApplicationBuilder"/></returns>
+        public static IFunctionsWorkerApplicationBuilder AddApplicationInsights(this IFunctionsWorkerApplicationBuilder builder, Action<ApplicationInsightsServiceOptions>? configureOptions = null)
+        {
+            builder.AddCommonServices();
+
+            builder.Services.AddApplicationInsightsTelemetryWorkerService(options =>
+            {
+                configureOptions?.Invoke(options);
+            });
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds the <see cref="ApplicationInsightsLoggerProvider"/> and disables the Functions host passthrough logger.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/></param>
+        /// <param name="configureOptions">Action to configure ApplicationInsights logger.</param>
+        /// <returns>The <see cref="IFunctionsWorkerApplicationBuilder"/></returns>
+        public static IFunctionsWorkerApplicationBuilder AddApplicationInsightsLogger(this IFunctionsWorkerApplicationBuilder builder, Action<ApplicationInsightsLoggerOptions>? configureOptions = null)
+        {
+            builder.AddCommonServices();
+
+            builder.Services.AddLogging(logging =>
+            {
+                logging.AddApplicationInsights(options =>
+                {
+                    options.IncludeScopes = false;
+                    configureOptions?.Invoke(options);
+                });
+            });
+
+            return builder;
+        }
+
+        private static IFunctionsWorkerApplicationBuilder AddCommonServices(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            builder.Services.TryAddEnumerable(new ServiceDescriptor(typeof(ITelemetryInitializer), typeof(FunctionsTelemetryInitializer), ServiceLifetime.Singleton));
+            builder.Services.TryAddEnumerable(new ServiceDescriptor(typeof(ITelemetryModule), typeof(FunctionsTelemetryModule), ServiceLifetime.Singleton));
+
+            // User logs will be written directly to Application Insights; this prevents duplicate logging.
+            builder.Services.AddSingleton<IUserLogWriter>(_ => NullUserLogWriter.Instance);
+
+            // This middleware is temporary for the preview. Eventually this behavior will move into the
+            // core worker assembly.
+            if (!builder.Services.Any(p => p.ImplementationType == typeof(FunctionActivitySourceMiddleware)))
+            {
+                builder.Services.AddSingleton<FunctionActivitySourceMiddleware>();
+                builder.Use(next =>
+                {
+                    return async context =>
+                    {
+                        var middleware = context.InstanceServices.GetRequiredService<FunctionActivitySourceMiddleware>();
+                        await middleware.Invoke(context, next);
```

Comment:
Yes, that's why we'll move this middleware to the Core bits before releasing. It won't actually be middleware in the end, we'll put it directly into the `FunctionsApplication.InvokeFunctionAsync()` so that it's (effectively) un-overrideable.